sklearn/ensemble/_forest.py

```diff
@@ -375,10 +375,18 @@ def fit(self, X, y, sample_weight=None):
             else:
                 sample_weight = expanded_class_weight
 
+        if not self.bootstrap and self.max_samples is not None:
+            raise ValueError(
+                "Sub-sample size only available if bootstrap=True. "
+                "Please leave max_samples to None."
```

Comment:
I would prefer to be specific regarding the name of the parameters in the error message

```suggestion
                "`max_sample` cannot be set if `bootstrap=False`. "
                "Either switch to `bootstrap=True` or set "
                "`max_sample=None`."
```